packages/dds/merge-tree/src/test/partialLength.spec.ts

```diff
@@ -0,0 +1,174 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { MergeTree } from "../mergeTree";
+import { MergeTreeDeltaType } from "../ops";
+import { PartialSequenceLengths } from "../partialLengths";
+import { TextSegment } from "../textSegment";
+import { insertText } from "./testUtils";
+
+describe("partial lengths", () => {
+    let mergeTree: MergeTree;
+    const localClientId = 17;
+    const remoteClientId = 18;
+
+    function validatePartialLengths(
+        clientId: number,
+        seq: number,
+        expectedLen?: number,
+    ): void {
+        const partialLen = mergeTree.root.partialLengths?.getPartialLength(
+            seq,
+            clientId,
+        );
+
+        let len = 0;
+
+        mergeTree.walkAllSegments(mergeTree.root, (segment) => {
+            if (segment.isLeaf() && !segment.removedClientIds?.length) {
```

Comment:
Logic now looks generally correct for checking if segment isn't removed in the view, but it doesn't verify that the segment has actually been inserted. This still passes the tests you have because they all only check partialLength at the latest seq. We should have some checks against older seq values still in the collab window (this doesn't need to be new tests--using existing ones and asserting over wider seq values should do it)

Reply:
still applies